test/loader-common.js

```diff
@@ -275,6 +290,71 @@ export const LoaderCommonFactory = (LoaderConstructor,
       assert.equal(progressEvents, 1, 'triggered progress');
     });
 
+    QUnit.test('aborts request at progress events if bandwidth is too low',
+    function(assert) {
+      const playlist1 = playlistWithDuration(10, { uri: 'playlist1.m3u8' });
+      const playlist2 = playlistWithDuration(10, { uri: 'playlist2.m3u8' });
+      const playlist3 = playlistWithDuration(10, { uri: 'playlist3.m3u8' });
+      const playlist4 = playlistWithDuration(10, { uri: 'playlist4.m3u8' });
+      const xhrOptions = {
+        timeout: 15000
+      };
+      let bandwidthupdates = 0;
+
+      playlist1.attributes.BANDWIDTH = 18000;
+      playlist2.attributes.BANDWIDTH = 10000;
+      playlist3.attributes.BANDWIDTH = 8888;
+      playlist4.attributes.BANDWIDTH = 7777;
+
+      loader.hls_.playlists = {
+        master: {
+          playlists: [
+            playlist1,
+            playlist2,
+            playlist3,
+            playlist4
+          ]
+        }
+      };
+
+      loader.on('bandwidthupdate', () => bandwidthupdates++);
+      loader.playlist(playlist1, xhrOptions);
+      loader.load();
+
+      this.clock.tick(1);
```

Comment:
If the order of each test here is unimportant (one test doesn't rely on another) you could break it up into a few smaller/more specific tests.  It might make it easier to debug if one specific case fails but the rest work.